src/components/validation/runner/mod.rs

```diff
@@ -573,23 +628,25 @@ fn spawn_output_driver(
 
             // Update the runner metrics for the received event. This will later
             // be used in the Validators, as the "expected" case.
-            let mut output_runner_metrics = output_runner_metrics.lock().unwrap();
+            let mut output_runner_metrics = output_runner_metrics.lock().await;
 
             for output_event in events {
-                output_runner_metrics.received_events_total += 1;
-                output_runner_metrics.received_event_bytes_total += vec![output_event.clone()]
-                    .estimated_json_encoded_size_of()
-                    .get()
-                    as u64;
-
-                if let Some(encoder) = maybe_encoder.as_ref() {
-                    let mut buffer = BytesMut::new();
-                    encoder
-                        .clone()
-                        .encode(output_event, &mut buffer)
-                        .expect("should not fail to encode output event");
-
-                    output_runner_metrics.received_bytes_total += buffer.len() as u64;
+                if !skip_output_driver_metrics {
+                    output_runner_metrics.received_event_bytes_total += vec![output_event.clone()]
```

Comment:
Why is this being cloned to get the estimated JSON size? And wrapped in a `Vec<Event>`, for that matter? :P

Reply:
The clone was added because of the need to wrap in the Vec.

I believe it has to do with the event stream for the actual events going through the topology. 

If I remove the wrap in Vec, then the actual vs expected comparison fails:

```
thread 'components::validation::tests::validate_component_tests_validation_components_sources_http_server_yaml' panicked at src/components/validation/mod.rs:352:25:
Failed to validate component 'http_server':
  test case 'happy path': failed
    - component_sent_event_bytes_total: expected 333, but received 339
  test case 'sad path': failed
    - component_sent_event_bytes_total: expected 222, but received 226
```

I suppose we could hard code that extra bytes but using the actual structures is more robust.

Reply:
I can add a comment explaining this since just by glancing it is not clear

Reply:
Now that I dig deeper, I see it's probably related to the fact that `Vec<T: EstimatedJsonEncodedSizeOf>` includes stuff like the `[]` brackets, and what not, in the estimation.

You should still theoretically be able to just put the event reference in the `Vec<T>`, though, since `EstimatedJsonEncodedSizeOf` is implemented for `Vec<V>` as `V: EstimatedJsonEncodedSizeOf`, and there's an impl of it for `&T` where `T: EstimatedJsonEncodedSizeOf`, etc.

Reply:
👍  Nice, was able to use a reference.